provider/baked/src/zerotrie.rs

```diff
@@ -111,17 +192,61 @@ impl<M: DataMarker> super::DataStore<M> for Data<M> {
     >;
     #[cfg(feature = "alloc")]
     fn iter(&'static self) -> Self::IterReturn {
-        #![allow(unused_imports)]
-        use alloc::borrow::ToOwned;
-        self.trie.iter().filter_map(move |(s, _)| {
-            if let Some((locale, attrs)) = s.split_once(ID_SEPARATOR as char) {
-                Some(DataIdentifierCow::from_owned(
-                    DataMarkerAttributes::try_from_str(attrs).ok()?.to_owned(),
-                    locale.parse().ok()?,
-                ))
-            } else {
-                s.parse().ok().map(DataIdentifierCow::from_locale)
-            }
-        })
+        iter(&self.trie)
+    }
+}
+
+pub struct DataForVarULEs<M: DataMarker>
+where
+    M::DataStruct: MaybeExportAsVarULE<'static>,
+{
+    // Unsafe invariant: actual values contained MUST be valid indices into `values`
+    trie: ZeroTrieSimpleAscii<&'static [u8]>,
+    values: &'static VarZeroSlice<<M::DataStruct as MaybeExportAsVarULE<'static>>::VarULE>,
+}
+
+impl<M: DataMarker> DataForVarULEs<M>
+where
+    M::DataStruct: MaybeExportAsVarULE<'static>,
+{
+    /// Construct from a trie and values
+    ///
+    /// # Safety
+    /// The actual values contained in the trie must be valid indices into `values`
+    pub const unsafe fn from_trie_and_values_unchecked(
+        trie: ZeroTrieSimpleAscii<&'static [u8]>,
+        values: &'static VarZeroSlice<<M::DataStruct as MaybeExportAsVarULE<'static>>::VarULE>,
+    ) -> Self {
+        Self { trie, values }
+    }
+}
+
+impl<M: DataMarker> super::DataStore<M> for DataForVarULEs<M>
+where
+    M::DataStruct: MaybeExportAsVarULE<'static>,
+    <M::DataStruct as MaybeExportAsVarULE<'static>>::VarULE: VarULE,
+    <M::DataStruct as MaybeExportAsVarULE<'static>>::EncodeAsVarULE:
+        From<&'static <M::DataStruct as MaybeExportAsVarULE<'static>>::VarULE>,
```

Comment:
why is there still a `From` bound?

Reply:
I need to transform `&str` (a `&Self::VarULE`) to `&str` (a `Self::EncodeAsVarULE`)

Reply:
The `From` bound is gone now. Will be replaced by ZeroFrom.